File: examples/other_app_having_app_links/src/main/java/com/example/other_app_having_app_links/MainActivity.java
package com.example.other_app_having_app_links;

import android.os.Bundle;
import android.webkit.WebView;

import androidx.appcompat.app.AppCompatActivity;

/**
 * other_app_having_app_links 作为第三方app，点击它的link，测试
 */
public class MainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        WebView webView = findViewById(R.id.web_view);
        webView.loadUrl("file:///android_asset/links_in_other_app.html");
    }
}

File: lib-examples/lifecycles-only/src/main/java/com/example/kotlin/lifecycle/v1/MainActivity.kt
package com.example.kotlin.lifecycle.v1

import android.os.Bundle
import android.widget.TextView
import androidx.appcompat.app.AppCompatActivity
import com.example.kotlin.lifecycle.R

/**
 * Update the count to screen
 */
class MainActivity : AppCompatActivity() {
    private lateinit var locationListener: LocationListener
    private lateinit var positionView: TextView

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)

        setContentView(R.layout.activity_main)
        positionView = findViewById(R.id.positionView)

        locationListener = LocationListener(this) { location: Location ->
            // Update UI
            location.position(object : Location.LocationCallback {
                override fun update(position: Int) {
                    runOnUiThread() {
                        positionView.text = position.toString()
                    }
                }
            })
        }
    }

    override fun onStart() {
        super.onStart()
        locationListener.start()
    }

    override fun onStop() {
        super.onStop()
        locationListener.stop()
    }
}
